tests/ui/app_list.js

```diff
@@ -125,21 +135,29 @@ appListPages.forEach(function(appListPage) {
                 // Test app src.
                 var href = this.getElementAttribute('.mkt-tile:nth-child(1)',
                                                     'href');
-                test.assert(href.indexOf('src=' + appListPage.src) !== -1,
-                           'Assert src');
+                if (appListPage.src) {
+                    test.assert(href.indexOf('src=' + appListPage.src) !== -1,
+                                'Assert src');
+                } else {
+                    // If no src is configured, it means this app list does not
+                    // contain any links.
+                    test.assertEqual(href, '', 'Assert href is empty');
+                }
 
                 // Test model cache.
-                var modelCount = casper.evaluate(function() {
-                    return Object.keys(
-                        window.require('models')('app')
-                              .data_store.app).length;
-                });
-                test.assertEqual(modelCount,
-                                 appListPage.collection ? 6 : APP_LIMIT,
-                                 'Assert model cache');
+                if (!appListPage.noModelCache) {
```

Comment:
I did a big refactor, moving a few tests out of the big one. This slows down testing (we have to reload the initial page list) but I think it's a bit easier to read/change now.